pkg/kubernetes/secrets/store.go

```diff
@@ -24,54 +24,60 @@ const (
 
 // Store implements the backing store for secrets as kubernetes secrets.
 type Store struct {
-	logger    hclog.Logger
-	config    config.Config
+	*portercontext.Context
 	hostStore cnabsecrets.Store
+	Secrets   map[string]map[string]string
+	namespace string
 	clientSet *kubernetes.Clientset
+	logger    hclog.Logger
 }
 
-func NewStore(cfg config.Config, l hclog.Logger) cnabsecrets.Store {
+func NewStore(c *portercontext.Context, cfg PluginConfig) *Store {
+	namespace := cfg.Namespace
 	s := &Store{
-		config:    cfg,
-		logger:    l,
-		hostStore: &host.SecretStore{},
+		Secrets:   make(map[string]map[string]string),
+		hostStore: &cnabhost.SecretStore{},
+		namespace: namespace,
+		logger:    cfg.Logger,
 	}
-
-	return portersecrets.NewSecretStore(s)
+	return s
 }
 
-func (s *Store) Connect() error {
+func (s *Store) connect() error {
 
 	if s.clientSet != nil {
 		return nil
 	}
-
-	clientSet, namespace, err := k8s.GetClientSet(s.config.Namespace, s.logger)
-
+	s.logger.Info(fmt.Sprintf("Store.connect: pre-clientset %s : %s", "namespace", s.namespace))
+	clientSet, namespace, err := k8shelper.GetClientSet(s.namespace)
 	if err != nil {
-		s.logger.Debug(fmt.Sprintf("Failed to get Kubernetes Client Set: %v", err))
 		return err
 	}
+	s.namespace = *namespace
+	s.logger.Info(fmt.Sprintf("Store.connect: post-clientset %s : %s", "namespace", s.namespace))
 
 	s.clientSet = clientSet
-	s.config.Namespace = *namespace
 
 	return nil
 }
 
 func (s *Store) Resolve(keyName string, keyValue string) (string, error) {
+	if err := s.connect(); err != nil {
+		return "", err
+	}
 	if strings.ToLower(keyName) != SecretSourceType {
 		return s.hostStore.Resolve(keyName, keyValue)
 	}
-
+	s.logger.Debug(fmt.Sprintf("Store.Resolve: ns:%s, keyName:%s, keyValue:%s", s.namespace, keyName, keyValue))
 	key := strings.ToLower(keyValue)
 
-	s.logger.Debug(fmt.Sprintf("Looking for key:%s", keyValue))
-	secret, err := s.clientSet.CoreV1().Secrets(s.config.Namespace).Get(context.Background(), key, metav1.GetOptions{})
+	secret, err := s.clientSet.CoreV1().Secrets(s.namespace).Get(context.Background(), key, metav1.GetOptions{})
 	if err != nil {
-		s.logger.Debug(fmt.Sprintf("Failed to Read secrets for key:%s %v", keyValue, err))
 		return "", err
 	}
-
-	return string(secret.Data[SecretDataKey]), nil
+	if val, ok := secret.Data[SecretDataKey]; !ok {
+		return "", InvalidSecretDataKeyError{msg: fmt.Sprintf("Key \"%s\" not found in secret", SecretDataKey)}
```

Comment:
I think it won't be clear enough what to do with this error message

Key credential not found in secret

How about we  provide a bit more context and what they should do to fix it?

The secret NAMESPACE/SECRET_NAME does not have a key named SECRETDATAKEY. The kubernetes.secrets plugin requires that the Kubernetes secret is named after the secret referenced in the Porter parameter or credential set, and secret value is stored in a key on the Kubernetes secret named SECRETDATAKEY.

Though I still think we need to also change the name of the secret data key since credential is a bit misleading when we are using the plugin to resolve parameter set secrets too.